lib/tilex_web/models/post.ex

```diff
@@ -80,6 +85,27 @@ defmodule Tilex.Post do
     |> hd
   end
 
+  def create_changeset(params, changes) do
+    %__MODULE__{}
+    |> cast(params, permitted_create_params())
+    |> validate_required(required_create_params())
+    |> validate_length(:title, max: title_max_chars())
+    |> validate_length_of_body
+    |> validate_number(:likes, greater_than: 0)
+    |> add_slug
+    |> change(changes)
+  end
+
+  def update_changeset(struct, params \\ %{}) do
```

Comment:
as a personal pref I'd remove the default value on `params` as we expect params to be sent every time when calling this func.